Mna 1695 configure linter

mna opened the pull request:
As discussed [here](https://github.com/fleetdm/fleet/pull/1769#pullrequestreview-736528708), having some more linters can help find issues by static analysis of the code, and this PR configures a relatively minimal set of linters with `golangci-lint` (https://golangci-lint.run/) - issues raised by those linters are on the safe side and not too opinionated. I fixed those that were raised, so this PR means that the `golangci-lint run` command succeeds.

For issues raised in existing DB migrations, I just added the `nolint` comment directive, to ignore the error, as there is no point to fix already applied migrations. I also did this (or similar) in a couple other places, I'll explain with inline comments.

That being said, as I didn't want to spend too much time on this (timeboxed ~ an hour), I haven't configured the Makefile to run it - there are a couple of tricky decisions to be made to add this there: the `golangci-lint` command must be installed and available, and it must work on Windows too - which I'm not too familiar on how to write makefiles conditions and shell calls that work properly there. However:

* There is a github action that exists to run the tool, so at least on CI we could easily add this check: https://github.com/golangci/golangci-lint-action.
* With Go 1.17, we can run a command without installing it first (https://golang.org/doc/go1.17#go-command, see the `go run` paragraph), so once we're comfortable requiring contributors to run Go 1.17, the `lint-go` makefile target could be as easy as `go run github.com/golangci/golangci-lint/cmd/golangci-lint@latest run`.

Happy to follow up with setting up the github action or work on makefile config if anyone has pointers on how to make it work immediately (without waiting for Go1.17).

# Checklist for submitter

If some of the following don't apply, please write a short explanation why.

- [ ] ~~Changes file added (if needed)~~ (I don't think this is needed for purely internal linting checks?)
- [ ] ~~Documented any API changes~~
- [ ] ~~Added tests for all functionality~~ (ran tests)
- [ ] ~~Manual QA for all functionality~~

mna commented on ee/fleetctl/updates.go:
```diff
@@ -29,7 +29,7 @@ const (
 	// https://github.com/theupdateframework/notary/blob/e87b31f46cdc5041403c64b7536df236d5e35860/docs/best_practices.md#expiration-prevention
 	// ~10 years
-	rootExpirationDuration = 10 * 365 * 24 * time.Hour
+	rootExpirationDuration = 10 * 365 * 24 * time.Hour //nolint:unused,deadcode
```
Even if not used, it seemed like we may want to keep it as part of the list of expiration constants.

mna commented on server/datastore/mysql/hosts_test.go:
```diff
@@ -351,58 +351,58 @@ func TestDeleteHost(t *testing.T) {
-	require.Nil(t, err)
-	require.Equal(t, hosts[0].ID, hosts2[0].ID)
-}
+//func testListHosts(t *testing.T, ds fleet.Datastore) {
```
Commented it out instead of removing it, could be meant to be an actual test? Otherwise we should probably delete it.

chiiph replied:
I deleted this in another PR, so +1 for removing it from my point of view.

mna commented on server/service/testing_utils.go:
```diff
@@ -25,12 +25,15 @@ func newTestService(ds fleet.Datastore, rs fleet.QueryResultStore, lq fleet.Live
 		fleetConfig.Filesystem.EnableLogRotation,
 		fleetConfig.Filesystem.EnableLogCompression,
 	)
+	//if err != nil {
```
If I actually add the check for the error, there's a failing test, so I opted to not check it now, but we should probably investigate and fix the test at some point.

chiiph replied:
Hm, interesting. And we should probably pass `t` here and `require.NoError` instead of `panic` whenever we address this.

zwass replied:
I created https://github.com/fleetdm/fleet/issues/1776 to track this.

mna commented:
Oh, I didn't realize we already had the golangci-lint running in CI (although only in the orbit/ directory). The failure is probably because of the older version used, I can update it and make it run on the whole repo if we're happy with the PR.

chiiph commented:
This looks great to me. @zwass where are you with `semgrep`? Is that something that would play along something like this, or should we define how we want to approach things before merging a PR like this one?

zwass reviewed (approved):
lgtm! Seems there is some issue running the `revive` linter?

Semgrep is something I'd like to add but it should happily exist alongside `golangci-lint`.

mna commented:
I updated the existing golangci-lint github action to use the latest version and to run on the whole repo instead of just orbit (and I renamed the file to reflect this).

Should I create an issue to track the changes that would be required to the `Makefile` if we want to replace `go vet ./...` of the `test-go` target with `golangci-lint`? Or are we fine with just having it run on CI, and manually as needed in dev (i.e. we take care of installing the tool and running it directly)? I guess the Makefile could also just assume that it is installed and call it as it calls e.g. yarn in other targets, and we could mention that `golangci-lint` is required in the "Building Fleet" docs.

mna commented:
> Should I create an issue to track the changes that would be required to the Makefile

I'll create one and document the various little impacts it would have on the existing github actions/workflows.

zwass commented:
I think adding `golangci-lint` (assuming that it is installed) to the `lint-go` target in the Makefile would be fine and then we can move towards using the new `go run` syntax soon (whenever Homebrew adds 1.17?)
